router/forwarder/tables.go

```diff
@@ -36,11 +37,13 @@ func newSessionTable() *sessionTable {
 }
 
 func (st *sessionTable) setForwardTable(sessionId string, ft *forwardTable) {
+	ft.last = time.Now()
 	st.sessions.Set(sessionId, ft)
 }
 
 func (st *sessionTable) getForwardTable(sessionId string) (*forwardTable, bool) {
 	if ft, found := st.sessions.Get(sessionId); found {
```

Comment:
This won't fix a garbage xgress with something stuck in retransmit, because the retransmitter will hit getForwardTable.

Could we key it off of the last time we got traffic from the xgress peer?

Reply:
It's not intended to fix that case.

Possibly. I'll look into it.

Reply:
Keying off xgress doesn't fix middle-hop routers in any way.